### PR TITLE
Gtm 1

### PR DESCRIPTION
**Pull request by anagarwa**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>

Test URLs:
- Before: https://main--icicidirect--aemsites.hlx.live/research/equity
- After: https://<branch>--icicidirect--aemsites.hlx.live/research/equity

- [ ] New Blocks introduced in this PR
      If yes, please provide details below

Block Name    | Documentation
------------- | -------------
 For e.g. carousel | [doc-link](https://main--icicidirect--aemsites.hlx.page/tools/sidekick/library.html?plugin=blocks&path=/sidekick/blocks/carousel&index=0)

### CODE REVIEW

File: scripts/delayed.js [hunk 1]
```diff
@@ -35,20 +35,22 @@ loadScript('https://icici-securities.allincall.in/files/deploy/embed_chatbot_11.
...
+        event: 'gtm.js',
+        'referrer': r // Pass the original referrer here
+    });
+    var f = d.getElementsByTagName(s)[0],
+        j = d.createElement(s), dl = l != 'dataLayer' ? '&l=' + l : '';
+    j.async = true;
+    j.src = 'https://www.googletagmanager.com/gtm.js?id=' + i + dl + '&referrer=' + encodeURIComponent(r);
+    console.log(j.src);
+    f.parentNode.insertBefore(j, f);
+}(window, document, 'script', 'dataLayer', 'GTM-WF9LTLZ', document.referrer));;
```
Comment (anagarwa) [inline scripts/delayed.js:51]: @sdmcraft  any thought on keeping the original referrer intact even with delay ?

File: scripts/delayed.js [hunk 2]
```diff
@@ -35,20 +35,22 @@ loadScript('https://icici-securities.allincall.in/files/deploy/embed_chatbot_11.
...
-        j.src =
-            'https://www.googletagmanager.com/gtm.js?id=' + i + dl;
-        f.parentNode.insertBefore(j, f);
-        }(window, document, 'script', 'dataLayer', 'GTM-WF9LTLZ'));
+        (function (w, d, s, l, i, r) {
+    w[l] = w[l] || [];
+    w[l].push({
+        'gtm.start': new Date().getTime(),
+        event: 'gtm.js',
+        'referrer': r // Pass the original referrer here
```
Comment (anagarwa) [inline scripts/delayed.js:43]: @sdmcraft any thought on keeping the original referrer intact with this solution even with delay ?

**Comment by JiangLong2019** [comment]
Maybe I'm missing the requirements of this ticket, but as I remember the `referrer` comes out-of-the-box in GTM which user can use {{referrer}} in the console. Do we have any specific use case that need to pass it by customization?

**Comment by anagarwa** [comment]
@JiangLong2019  sorry I was trying to do some testing , this PR was not ready for the review yet. I have updated the draft lablel
Passing referrer in the method was to sure referrer does not change basis on future event in case of lazy loading the script

**Comment by JiangLong2019** [comment]
@anagarwa Got it, thanks

**Pull request closed without merging**